exec/java-exec/src/main/java/org/apache/drill/exec/util/FileSystemUtil.java

```diff
@@ -302,12 +332,32 @@ protected List<FileStatus> compute() {
       List<RecursiveListing> tasks = new ArrayList<>();
 
       try {
-        for (FileStatus status : fs.listStatus(path, filter)) {
+        FileStatus[] dirFs = fs.listStatus(path, filter);
+        if (recursiveListingMaxSize > 0 && fileCounter.addAndGet(dirFs.length) > recursiveListingMaxSize) {
+          throw UserException
```

Comment:
This code is executed within fork join pool, and if error suppression flag is enabled, it will call task.fork(). Can we somehow stop executing all tasks for the case when count is reached?

Reply:
@vvysotskyi I've added an attempt to do that now.